packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -930,7 +948,15 @@ export class ContainerRuntime
 	 * @param callback - the callback to be invoked
 	 */
 	public ensureNoDataModelChanges<T>(callback: () => T): T {
-		this.ensureNoDataModelChangesCalls++;
+		if (this.ensureNoDataModelChangesCalls++ > 100) {
```

Comment:
It not having this code results in stack overflow, then I'd rather remove it - I do not think we are getting better error reporting (if I'm wrong, then I'd add a note in the code describing why it's beneficial to have this code)

Reply:
I would actually decrease the limit to 50 as well